chain.go

```diff
@@ -350,23 +367,24 @@ func (c *Chain) Walk(fn WalkerFn) (err error) {
 // Validate traverses chain confirming the hashes
 // @TODO confirm that TypeLinks are also correct
 // @TODO confirm signatures
-func (c *Chain) Validate(h HashSpec) (err error) {
-	l := len(c.Headers)
+// Does a Holochain have a one-to-one with a Chain? We shouldn't need both
+func (holo *Holochain) ValidateChain() (err error) {
+	l := len(holo.chain.Headers)
 	for i := l - 1; i >= 0; i-- {
-		hd := c.Headers[i]
+		hd := holo.chain.Headers[i]
 		var hash Hash
 
 		// hash the header
-		hash, _, err = hd.Sum(h)
+		hash, _, err = hd.Sum(holo)
```

Comment:
I don't like these method patterns, what do you think about using the hash[, err] = holo.Sum() method pattern when we need the holochain?